src/main/java/org/cyclonedx/util/deserializer/ToolDeserializer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * This file is part of CycloneDX Core (Java).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) OWASP Foundation. All Rights Reserved.
+ */
+package org.cyclonedx.util.deserializer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.cyclonedx.model.Tool;
+
+import java.io.IOException;
+
+public class ToolDeserializer
+    extends JsonDeserializer<Tool>
+{
+  private final ObjectMapper mapper = new ObjectMapper();
+
+  @Override
+  public Tool deserialize(JsonParser parser, DeserializationContext context) throws IOException {
+    ObjectCodec codec = parser.getCodec();
+    JsonNode node = codec.readTree(parser);
+    return mapper.convertValue(node, Tool.class);
```

Comment:
without this does not parse the tool? I don't see any logic

Reply:
> without this does not parse the tool? 
Nope. Throws a null pointer.

> I don't see any logic
I think the mapper is smart enough to use the class definition to how to deserialize? Similar to [here](https://github.com/shaikhu/cyclonedx-core-java/blob/691649c4d0f7adeba4a27ddb1ddf33a69f0c9c1b/src/main/java/org/cyclonedx/util/deserializer/SignatoryDeserializer.java#L74..L75).